### DISCUSSION TITLE
[macOS] (bug) Font rendering changes dramatically on external monitor

### DISCUSSION THREAD

**Discussion by charlesnicholson** [ghostty-org/ghostty#4197] [Q&A] [answered]
Hi everyone-

I'm on a 2023 16" MacBook Pro M2 16" running Sequoia (15.2), connected to an external 24" HP non-4k display. I set up ghostty over the weekend and it looks great on the laptop's native Retina display, but then when I dock my laptop and run ghostty on my external display, it becomes very ugly- the font spacing becomes wrong. Additionally, there's a noticeable and ugly "pop" effect that happens, though I don't really care about that- I'm just mentioning it for completeness.

This feels more like a bug than a configuration issue- here are the details.

My config is simple:
```
theme = Kanagawa Wave

font-family = Symbols Nerd Font Mono
font-family = Hack Nerd Font
font-size = 12

shell-integration = zsh
window-save-state = always
bold-is-bright = true
link-url = true

macos-auto-secure-input = true
macos-secure-input-indication = true

quit-after-last-window-closed = true
```

Here's what ghostty looks like on my retina display:
<img width="798" alt="Screenshot 2024-12-31 at 9 54 20 AM" src="https://github.com/user-attachments/assets/6a26d66c-07a2-4c1c-bbb3-1beab7539875" />

and here's what the same window looks like when I drag it to my external monitor:
![Screenshot 2024-12-31 at 9 54 08 AM](https://github.com/user-attachments/assets/f2b94673-298a-406d-adb6-cf46549a20c8)

(The differences become much more visible when you click on the images to see them at their native resolutions, though you can kinda see it inlined here as well).

Here's a movie of me dragging the window between the two monitors (Quicktime can't record both monitors at once so the capture is from my external monitor):

https://github.com/user-attachments/assets/e19ab4c7-f041-4192-ada3-30201c69aa0e

As a baseline comparison, iTerm2 does not pop or change the font spacing when it's dragged between monitors, and looks identical (to my eye) on both displays.

TIA for reading and any suggestions!

**Comment by charlesnicholson** [comment]
For completeness, I should also mention that the rendering is different for ghostty windows that I open immediately on either monitor. It's not a function of actually dragging an existing window across the monitor boundaries.

**Comment by charlesnicholson** [comment, marked as answer]
For anyone who finds this via search, it's part of this bug: https://github.com/ghostty-org/ghostty/issues/4504

**Comment by erf** [comment]
I recently noticed that on my Macbook Air, the emojis are left aligned like here:
<img width="52" alt="Screenshot 2025-07-03 at 19 19 59" src="https://github.com/user-attachments/assets/ca45e3d4-1816-43b1-a5cb-e0e50456a299" />

But on my external monitor they are right aligned and bleeding one pixel to the right.
<img width="52" alt="Screenshot 2025-07-03 at 19 19 59" src="https://github.com/user-attachments/assets/50426081-4a51-4251-9501-1a1999cff151" />

**Reply by taylrfnt** [reply]
Please don't try and revive posts from 6+ mo ago, instead create a new discussion with your issue's information.

Edit: Looks like you already have - #3675

**Reply by erf** [reply]
Ok. My other post is also old and maybe unrelated. Do you suggest i create a new discussion?

**Reply by taylrfnt** [reply]
I think your original post is relevant enough, no need for a new post.  The old one has popped back into the recent view since you've add the same comment there, so if maintainers have thoughts or others have suggestions they'll respond.  Text rendering is just not my area of expertise otherwise I would try and push discussion further along, sorry.
